src/components/map-controls.js

```diff
@@ -72,24 +71,46 @@ export default class MapControls extends PureComponent {
   }
 
   componentDidMount() {
-    // Register event handlers on the canvas using the EventManager helper class
-    //
-    // Note that mouse move and click are handled directly by static-map
-    // Corresponding to hover and click on map
-    // onMouseMove={this._onMouseMove}
-    // onMouseClick={this._onMouseClick}
-
-    this._eventManager = new EventManager(this.refs.canvas, {
-      onMouseDown: this._onMouseDown,
-      onMouseDrag: this._onMouseDrag,
-      onMouseUp: this._onMouseUp,
-      onTouchRotate: this._onTouchRotate,
-      onWheel: this._onWheel,
-      onWheelEnd: this._onWheelEnd,
-      mapTouchToMouse: true
+    // Register event handlers
+    const {canvas} = this.refs;
+
+    this._eventManager = new EventManager(canvas)
+    .on({
+      panstart: this._onDragStart,
+      pan: this._onDrag,
+      panend: this._onDragEnd,
+      pinchstart: this._onPinchStart,
+      pinch: this._onPinch,
+      pinchend: this._onPinchEnd,
+      doubletap: this._onDoubleTap,
+      wheel: this._onWheel
     });
   }
 
+  componentWillUnmount() {
+    if (this._eventManager) {
+      // Must destroy because hammer adds event listeners to window
+      this._eventManager.destroy();
+    }
+  }
+
+  /* Event utils */
```

Comment:
The properties used here are all from the hammer.js [event object](http://hammerjs.github.io/api/#event-object). Will add to comment.